extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authentication/db/cache/CoordinatorBasicAuthenticatorCacheNotifier.java

```diff
@@ -0,0 +1,280 @@
+
+        // best effort, if this fails, remote node will poll and pick up the update eventually
+        Request req = new Request(HttpMethod.POST, listenerURL);
+        req.setContent(MediaType.APPLICATION_JSON, serializedUserAndRoleMap);
+
+        BasicAuthDBConfig authenticatorConfig = authenticatorConfigMap.get(updatedAuthenticatorPrefix);
+
+        ListenableFuture<StatusResponseHolder> future = httpClient.go(
+            req,
+            new ResponseHandler(),
+            Duration.millis(authenticatorConfig.getCacheNotificationTimeout())
+        );
+        futures.add(future);
+      }
+    }
+    return futures;
+  }
+
+  private void initAuthenticatorConfigMap(AuthenticatorMapper mapper)
+  {
+    if (mapper == null || mapper.getAuthenticatorMap() == null) {
+      return;
+    }
+
+    for (Map.Entry<String, Authenticator> entry : mapper.getAuthenticatorMap().entrySet()) {
+      Authenticator authenticator = entry.getValue();
+      if (authenticator instanceof BasicHTTPAuthenticator) {
+        String authenticatorName = entry.getKey();
+        BasicHTTPAuthenticator basicHTTPAuthenticator = (BasicHTTPAuthenticator) authenticator;
+        BasicAuthDBConfig dbConfig = basicHTTPAuthenticator.getDbConfig();
+        authenticatorConfigMap.put(authenticatorName, dbConfig);
+      }
+    }
+  }
+
+  private static URL getListenerURL(DruidNode druidNode, String authPrefix)
+  {
+    try {
+      return new URL(
+          druidNode.getServiceScheme(),
+          druidNode.getHost(),
+          druidNode.getPortToUse(),
+          StringUtils.format("/druid-ext/basic-security/authentication/listen/%s", authPrefix)
+      );
+    }
+    catch (MalformedURLException mue) {
+      LOG.error("WTF? Malformed url for DruidNode[%s] and authPrefix[%s]", druidNode, authPrefix);
+      throw new RuntimeException(mue);
+    }
+  }
+
+  // Based off StatusResponseHandler, but with response content ignored
+  private static class ResponseHandler implements HttpResponseHandler<StatusResponseHolder, StatusResponseHolder>
+  {
+    @Override
+    public ClientResponse<StatusResponseHolder> handleResponse(HttpResponse response)
+    {
+      return ClientResponse.unfinished(
+          new StatusResponseHolder(
+              response.getStatus(),
+              null
+          )
+      );
+    }
+
+    @Override
+    public ClientResponse<StatusResponseHolder> handleChunk(
+        ClientResponse<StatusResponseHolder> response,
+        HttpChunk chunk
+    )
+    {
+      return response;
+    }
+
+    @Override
+    public ClientResponse<StatusResponseHolder> done(ClientResponse<StatusResponseHolder> response)
+    {
+      return ClientResponse.finished(response.getObj());
+    }
+
+    @Override
+    public void exceptionCaught(
+        ClientResponse<StatusResponseHolder> clientResponse, Throwable e
+    )
+    {
+      // Its safe to Ignore as the ClientResponse returned in handleChunk were unfinished
```

Comment:
Would you elaborate on this?

Reply:
This was copied from `FullResponseHandler`